lib/index.js

```diff
@@ -1,16 +1,83 @@
-function updateWeather(cityState) {
+const initialVisit = () => {
+  if(document.cookie) {
+    let cookies = document.cookie.split('; ');
+    apiKey = cookies[0].split('=')[1];
+    let username = cookies[1].split('=')[1];
+    if(apiKey != '') {
+      let usernameElementParent = document.getElementById('hello-user');
+      let usernameElement = document.createElement('p');
+      let usernameText = document.createTextNode('Welcome, ' + username);
+      usernameElement.appendChild(usernameText);
+      usernameElementParent.appendChild(usernameElement);
+      document.getElementById('login-link').classList.add('hidden');
+      document.getElementById('register-link').classList.add('hidden');
+      document.getElementById('see-favorites-link').classList.remove('hidden');
+      document.getElementById('logout-link').classList.remove('hidden');
+    } else {
+      document.getElementById('login-link').classList.remove('hidden');
+      document.getElementById('register-link').classList.remove('hidden');
+      document.getElementById('see-favorites-link').classList.add('hidden');
+      document.getElementById('logout-link').classList.add('hidden');
+    }
+  }
+}
+
+let apiKey = '';
+initialVisit();
+
+function updateWeather(cityState, cityStateName) {
   $.getJSON(`https://sweater-weather-1810.herokuapp.com/api/v1/forecast?location=${cityState}`, function(response){
     let weatherData = response.data;
-    updateCurrentWeather(weatherData);
+    clearDOM();
+    updateCurrentWeather(weatherData, cityStateName);
```

Comment:
Ya, like this, separate out what sets the weather components from the event that is calling it. In a lot of the JS frameworks there is a concept of a component, which is basically a small piece of you app, say the weather display, which contains all of the methods and styles used for that piece.